### PR TITLE
Windows common fixes and document offering

### PR DESCRIPTION
**Pull request by rs-sam**
Windows backend and its interface offering document.
General complier specific warning issue fix.

### CODE REVIEW

File: .reuse/dep5 [hunk 1]
```diff
@@ -43,10 +43,6 @@ Files: toolbox/cijoe-pkg-xnvme/*
 Copyright: Samsung Electronics Co., Ltd
 License: BSD-3-Clause
 
-Files: third-party/windows/getopt/*/*
```
Comment (safl) [inline .reuse/dep5:46]: Great.

File: lib/xnvme_be_windows_async_ioring.c [hunk 2]
```diff
@@ -5,14 +5,14 @@
 #ifndef _XOPEN_SOURCE
 #define _XOPEN_SOURCE 700
 #endif
+#ifdef XNVME_BE_WINDOWS_ASYNC_ENABLED
```
Comment (safl) [inline lib/xnvme_be_windows_async_ioring.c:8]: LGTM.

File: docs/getting_started/toolchain.rst [hunk 3]
```diff
@@ -874,9 +876,9 @@ Or, run the commands contained within the script manually:
 
 
 
-Then go ahead and configure, build and install using ``meson``:
+Then go ahead and configure, build and install using helper batch script ``build.bat``:
 
-.. literalinclude:: ../../toolbox/pkgs/default-build.sh
+.. literalinclude:: build_windows.rst
```
Comment (safl) [inline docs/getting_started/toolchain.rst:881]: Excellent!

**Review by safl — CHANGES_REQUESTED** [review]
Just one minor thing. Otherwise, great!

Forgot this, please add ``#include <inttypes.h>`` and use the PRIu64 / PRIx64 etc. Macros for print-format-specifiers, since changing them to e.g. ``%llu`` will yield errors for clang / gcc.

**Review by karlowich — CHANGES_REQUESTED** [review]
Looks good!
I have added a few comments

File: docs/getting_started/index.rst [hunk 4]
```diff
@@ -372,6 +372,126 @@ filepaths::
...
+* ``IOCTL_STORAGE_SET_PROPERTY``
+* ``IOCTL_STORAGE_REINITIALIZE_MEDIA``
+* ``IOCTL_SCSI_PASS_THROUGH_DIRECT``
+
+You can check that this interface is behaving as expected by running:
+
+.. literalinclude:: xnvme_win_info_default.cmd
+   :language: bash
+
+Which you yield output equivalent to:
```
[the remark this answers is not in the document]
Comment (rs-sam) [reply docs/getting_started/index.rst]: taken

File: docs/getting_started/index.rst [hunk 5]
```diff
@@ -372,6 +372,126 @@ filepaths::
...
+
+.. literalinclude:: xnvme_win_info_default.out
+   :language: bash
+   :lines: 1-12
+
+This tells you that **xNVMe** can communicate with the given device identifier
+and it informs you that it utilizes **nvme_ioctl** for synchronous command
+execution and it uses **iocp** for asynchronous command execution. This method
+can be used for raw devices via **\\.\PhysiaclDrive<disk number>** device path.
```
[the remark this answers is not in the document]
Comment (rs-sam) [reply docs/getting_started/index.rst]: fixed

File: docs/getting_started/index.rst [hunk 6]
```diff
@@ -372,6 +372,126 @@ filepaths::
...
+and it informs you that it utilizes **nvme_ioctl** for synchronous command
+execution and it uses **iocp** for asynchronous command execution. This method
+can be used for file operations via **<driver name>:\<file name>** path.
+
+Async I/O via ``iocp``
+~~~~~~~~~~~~~~~~~~~~~~
+
+When AIO is available then the NVMe NVM Commands for **read** and **write** are
+sent over the Windows IOCP interface. Doing so improves command-throughput at
+higher queue-depths when compared to sending the command over via the NVMe
```
[the remark this answers is not in the document]
Comment (rs-sam) [reply docs/getting_started/index.rst]: fixed

File: docs/getting_started/build_windows.rst [hunk 7]
```diff
@@ -3,5 +3,14 @@
...
+  build.bat
+
+  # config: debug only configure xNVMe
+  build.bat config-debug
+
+  # install xNVMe
+  build.bat install
+
+  # uninstall xNVMe
+  build.bat uninstall
```
[the remark this answers is not in the document]
Comment (rs-sam) [reply docs/getting_started/build_windows.rst]: fixed

File: docs/getting_started/index.rst [hunk 8]
```diff
@@ -372,6 +372,126 @@ filepaths::
...
+
+.. literalinclude:: xnvme_win_io_async_read_iocp.out
+   :language: bash
+   :lines: 1-12
+
+
+Async I/O via ``iocp-th``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Similar to ``iocp`` interface, only difference is seperate poller is used
```
[the remark this answers is not in the document]
Comment (rs-sam) [reply docs/getting_started/index.rst]: fixed

**Review by karlowich — APPROVED** [review]
Great!

**Review by safl — APPROVED** [review]

**Pull request merged**